### DISCUSSION TITLE
Update from 6.1.0 to 6.2.0

### DISCUSSION THREAD

**Discussion by diderich** [YetiForceCompany/YetiForceCRM#14889] [General]
I just upgraded by CRM system from version 6.1.0 v8 to 6.2.0 v1 and want to share my experience.

First, I had not major issues with the automatic upgrade. Nevertheless I observed the following issues and had to make the following changes:

1. I got the following error messages in my logfile. They did not pose any installation problem as identified so far.
```
[Sat Jul 03 09:41:05.242418 2021] [php7:warn] [pid 23991] [client xxxx.xxxx.xxxx:xxxxx PHP Warning:  print_r() expects at most 2 parameters, 4 given in /opt/crm6/cache/updates/updates/init.php on line 650
[Sat Jul 03 09:41:33.596017 2021] [php7:error] [pid 23991] [client xxxx.xxxx.xxxx:xxxxx] PHP Fatal error:  Declaration of App\\YetiForce\\Shop\\Product\\YetiForceVulnerabilities::verify(): array must be compatible with App\\YetiForce\\Shop\\AbstractBaseProduct::verify(): bool in /opt/crm6/app/YetiForce/Shop/Product/YetiForceVulnerabilities.php on line 41
```
2. I had to update/delete some customized files in the `custom` directory, that were not longer consistent with version 6.2.
3. I did not succeed to update/install roundcube (see [https://github.com/YetiForceCompany/YetiForceCRM/issues/14887](https://github.com/YetiForceCompany/YetiForceCRM/issues/14887)). This is not a major issue for me as I don't use roundcube for e-mail.
4. I had to update some of the files in `module/XXX/summary_block`, as the functionality has been changed/updated (in a way that by the way I like).
5. I had to change some parameters in `config/Security.php` to disable having to enter a reason why switching user.

If I find more issues/changes, I will document them here so that other users can learn from my experience.

**Comment by diderich** [comment]
6. If using `\App\Fields\RangeTime::displayElapseTime` to display the time in days / hours / minutes, a day is considered 24 hours. I have added a function `\App\Fields\RangeTime::displayWorkTime` that calculates a day as 8 hours, rather than 24, making the display I use in `summary_blocks` more sensible.

**Reply by mariuszkrzaczkowski** [reply]
share the "displayWorkTime" function

**Reply by diderich** [reply]
```php
	/**
	 * Format elapsed time to short display value.
	 *
	 * @param float    $decTime     time in decimal format (in minutes)
	 * @param string   $type        text format 'short' or 'full' (for label)
	 * @param bool|int $withSeconds if is provided as int then will be displayed
	 *
	 * @return string
	 */
	public static function displayWorkTime($decTime, $type = 'short', $withSeconds = false)
	{
		$short = 'short' === $type;
		$decTime = $decTime / 60;

		$day = floor($decTime / 8);
		$hour = floor($decTime - 8 * $day);
		$min = floor(($decTime - $hour - 8 * $day) * 60);
		$sec = round((($decTime - $hour - 8 * $day) * 60 - $min) * 60);

		$result = '';
		if ($day > 0) {
			$result .= $short ? $day . \App\Language::translate('LBL_D') : "{$day} " . \App\Language::translate('LBL_DAYS');
		}
		if ($hour) {
			$result .= $short ? " {$hour}" . \App\Language::translate('LBL_H') : "{$hour} " . \App\Language::translate('LBL_HOURS');
		}
		if ($hour || $min) {
			$result .= $short ? " {$min}" . \App\Language::translate('LBL_M') : " {$min} " . \App\Language::translate('LBL_MINUTES');
		}
		if (false !== $withSeconds) {
			$result .= $short ? " {$sec}" . \App\Language::translate('LBL_S') : " {$sec} " . \App\Language::translate('LBL_SECONDS');
		}
		if (!$hour && !$min && false === $withSeconds) {
			$result = $short ? '0' . \App\Language::translate('LBL_M') : '0 ' . \App\Language::translate('LBL_MINUTES');
		}
		return trim($result);
	}

```

**Comment by diderich** [comment]
7. As the meaning of the field `vtiger_osstimecontrol.sum_time` has changed its definition, from hours to minutes, I had to update various `app/TextParser` code segments that I have programmed to generate customized reports as well as time worked calculations in `module/XXX/summary_block`. This is only relevant to you, if you have configured/changed/added code yourself.
